monai/handlers/logfile_handler.py

```diff
@@ -0,0 +1,85 @@
+# limitations under the License.
+
+import logging
+import os
+from typing import TYPE_CHECKING, Optional
+
+from monai.config import IgniteInfo
+from monai.utils import min_version, optional_import
+
+Events, _ = optional_import("ignite.engine", IgniteInfo.OPT_IMPORT_VERSION, min_version, "Events")
+if TYPE_CHECKING:
+    from ignite.engine import Engine
+else:
+    Engine, _ = optional_import("ignite.engine", IgniteInfo.OPT_IMPORT_VERSION, min_version, "Engine")
+
+
+__all__ = ["LogfileHandler"]
+
+
+class LogfileHandler:
+    """
+    Adds a `logging.FileHandler` to the attached engine's logger when the start event occurs and removes it again when
+    then completed event occurs.
+
+    A handler is needed to remove `FileHandler` object when the complete event occurs so that further runs of different
+    engines write only to the log files they should, rather than previous files. Multiple handlers can write to the same
+    file which allows output from train and evaluation engine objects to be condensed in one file. If the given output
+    directory doesn't exist it will by default be created when the  start event occurs. This can be used in conjunction
+    with `CheckpointSaver` to save a log file to the same destination as the saved checkpoints. Since the handler is
+    added possibly after other logging events during initialisation, not all logging data will be retained.
+
+    Args:
+        output_dir: directory to save the log file to
+        filename: name of the file to save log to
+        loglevel: log level for the handler
+        formatter: format string for the `logging.Formatter` set for the handler
+        create_dir: if True, create `output_dir` if it doesn't exist
+    """
+
+    def __init__(
+        self,
+        output_dir: str,
+        filename: str = "log.txt",
+        loglevel: int = logging.INFO,
+        formatter: str = "%(asctime)s %(name)s %(levelname)s: %(message)s",
+        create_dir: bool = True,
+    ):
+        self.output_dir: str = output_dir
+        self.filename: str = filename
+        self.loglevel: int = loglevel
+        self.formatter: str = formatter
+        self.create_dir: bool = create_dir
+        self.logger: Optional[logging.Logger] = None
+        self.handler: Optional[logging.FileHandler] = None
+
+    def attach(self, engine: Engine) -> None:
+        self.logger = engine.logger
+        engine.add_event_handler(Events.STARTED, self._start)
+        engine.add_event_handler(Events.COMPLETED, self._completed)
+
+    def _start(self, engine: Engine) -> None:
+        if self.create_dir and not os.path.exists(self.output_dir):
+            os.makedirs(self.output_dir, exist_ok=True)
+
+        self.handler = logging.FileHandler(os.path.join(self.output_dir, self.filename))
+        self.handler.setLevel(self.loglevel)
+        self.handler.setFormatter(logging.Formatter(self.formatter))
+        assert self.logger is not None
+        self.logger.addHandler(self.handler)
```

Comment:
Hi @wyli @ericspod ,

I remember we discussed the policy of stats handler, we should leverage the standard way of python logging:
Let the users to set log handler (not ignite handler) to the logger outside of the `StatsHandler`, then we can easily log info to files or console:
https://github.com/Project-MONAI/MONAI/blob/dev/monai/handlers/stats_handler.py#L114
So we removed all the `handler` arg in ignite handlers, suppose to use `logging.getLogger(name)` to control it.
What do you think?

Thanks.

Reply:
I was thinking that different engines might end up with different logger objects so doing it this way ensured you got the right one, it would be compatible with vanilla Ignite engine objects. The problem I encountered was to do the log file handling in bundles wasn't straight forward.